features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/Environments.jsx

```diff
@@ -57,188 +272,1561 @@ const useStyles = makeStyles((theme) => ({
+    const [description, setDescription] = useState('');
+    const [mgLabels, setMgLabels] = useState([]);
+    const [SelectedEnvironment, setSelectedEnvironment] = useState([]);
+    const [open, setOpen] = useState(false);
+    const [confirmDeleteOpen, setConfirmDeleteOpen] = useState(false);
+    const [revisionToDelete, setRevisionToDelete] = useState([]);
+    const [confirmRestoreOpen, setConfirmRestoreOpen] = useState(false);
+    const [revisionToRestore, setRevisionToRestore] = useState([]);
+    const [openDeployPopup, setOpenDeployPopup] = useState(false);
+    const [lastRevisionCount, setLastRevisionCount] = useState(0);
+
+    const extractLastRevisionNumber = (list, lastRev) => {
+        if (lastRev !== null) {
+            setLastRevisionCount(lastRev);
+        } else if (list[list.length - 1]) {
+            const lastRevisionName = list[list.length - 1].displayName;
+            const splitList = lastRevisionName.split(' ');
+            setLastRevisionCount(splitList[1]);
+        } else {
+            setLastRevisionCount(0);
+        }
+    };
+
     useEffect(() => {
         restApi.getDeployments()
             .then((result) => {
                 setAllDeployments(result.body.list);
             });
+        restApi.microgatewayLabelsGet()
+            .then((result) => {
+                setMgLabels(result.body.list);
+            });
+        restApi.getRevisions(api.id).then((result) => {
+            setRevisions(result.body.list);
+            setLastRevisionCount(result.body.count);
+            extractLastRevisionNumber(result.body.list, null);
+        });
+        restApi.getRevisionsWithEnv(api.isRevision ? api.revisionedApiId : api.id).then((result) => {
+            setEnvRevision(result.body.list);
+        });
     }, []);
 
+    const toggleOpenConfirmDelete = (revisionName, revisionId) => {
+        setRevisionToDelete([revisionName, revisionId]);
+        setConfirmDeleteOpen(!confirmDeleteOpen);
+    };
+
+    const toggleOpenConfirmRestore = (revisionName, revisionId) => {
+        setRevisionToRestore([revisionName, revisionId]);
+        setConfirmRestoreOpen(!confirmRestoreOpen);
+    };
+
+    const toggleDeployRevisionPopup = () => {
+        setOpenDeployPopup(!openDeployPopup);
+    };
+
+    const handleCloseDeployPopup = () => {
+        setOpenDeployPopup(false);
+        setExtraRevisionToDelete(null);
+    };
+
+    const handleClickOpen = () => {
+        setOpen(true);
+    };
+
+    const handleDeleteSelect = (event) => {
+        setExtraRevisionToDelete([event.target.value, event.target.name]);
+    };
+
+    const handleSelect = (event) => {
+        setRevision(event.target.value);
+    };
+
+    const handleClose = () => {
+        setOpen(false);
+        setExtraRevisionToDelete(null);
+    };
+
+    function checkIfDeletingLastRevision(list, revisionName) {
+        const splitList = revisionName.split(' ');
+        let splitList1;
+        if (list[list.length - 1]) {
```

Comment:
We are running the same logic to exact the next revision number in `extractLastRevisionNumber` shall we move this to a util method?
and also if this revision naming patter gets changed, this logic will break `Revision xxx`

@CrowleyRajapakse Can't we send the revision id kind of thing in the rest API response ?